internal/handler/agent_middleware_test.go

```
@@ -0,0 +1,267 @@
+// =============================================================================
+
+func TestBootstrapRedirectMiddleware_EmptyBootstrapHost_PassesThrough(t *testing.T) {
+	// bootstrapHost="" disables the middleware entirely. Verify
+	// the returned handler is the same one we passed in (no
+	// wrapping) — this is what makes the middleware safe to mount
+	// unconditionally even when bootstrap isn't configured.
+	inner := newOKHandler()
+	got := BootstrapRedirectMiddleware(inner, "", "anything.example.com", newTestLogger())
+
+	rec := httptest.NewRecorder()
+	req := httptest.NewRequest(http.MethodGet, "http://whatever.example.com/path", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)